go/common/node/sgx_test.go

```diff
@@ -0,0 +1,131 @@
+package node
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/oasisprotocol/oasis-core/go/common/cbor"
+)
+
+func TestSGXConstraintsV0(t *testing.T) {
+	require := require.New(t)
+
+	raw, err := ioutil.ReadFile("testdata/sgx_constraints_v0.bin")
+	require.NoError(err, "Read test vector")
+
+	var sc SGXConstraints
+	err = cbor.Unmarshal(raw, &sc)
+	require.NoError(err, "Decode V0 SGX constraints")
+
+	err = sc.ValidateBasic(nil)
+	require.NoError(err, "ValidateBasic V0 SGX constraints")
+
+	enc := cbor.Marshal(sc)
+	require.EqualValues(enc, raw, "serialization should round-trip")
+}
+
+func TestSGXConstraintsV1(t *testing.T) {
+	require := require.New(t)
+
+	sc := SGXConstraints{
+		Versioned: cbor.NewVersioned(1),
+	}
+	err := sc.ValidateBasic(nil)
+	require.Error(err, "ValidateBasic V1 SGX constraints without PCS support")
+	err = sc.ValidateBasic(&TEEFeatures{})
+	require.Error(err, "ValidateBasic V1 SGX constraints without PCS support")
+	err = sc.ValidateBasic(&TEEFeatures{SGX: TEEFeaturesSGX{PCS: true}})
+	require.NoError(err, "ValidateBasic V1 SGX constraints")
+}
+
+func TestSGXAttestationV0(t *testing.T) {
+	require := require.New(t)
+
+	raw, err := ioutil.ReadFile("testdata/sgx_attestation_v0.bin")
+	require.NoError(err, "Read test vector")
+
+	var sa SGXAttestation
+	err = cbor.Unmarshal(raw, &sa)
+	require.NoError(err, "Decode V0 SGX attestation")
+
+	err = sa.ValidateBasic(nil)
+	require.NoError(err, "ValidateBasic V0 SGX attestation")
+
+	enc := cbor.Marshal(sa)
+	require.EqualValues(enc, raw, "serialization should round-trip")
+}
+
+func TestSGXAttestationV1(t *testing.T) {
+	require := require.New(t)
+
+	sa := SGXAttestation{
+		Versioned: cbor.NewVersioned(1),
+	}
+	err := sa.ValidateBasic(nil)
+	require.Error(err, "ValidateBasic V1 SGX attestation without PCS support")
+	err = sa.ValidateBasic(&TEEFeatures{})
+	require.Error(err, "ValidateBasic V1 SGX attestation without PCS support")
+	err = sa.ValidateBasic(&TEEFeatures{SGX: TEEFeaturesSGX{PCS: true}})
+	require.NoError(err, "ValidateBasic V1 SGX attestation")
+
+	raw, err := ioutil.ReadFile("testdata/sgx_attestation_v1.bin")
+	require.NoError(err, "Read test vector")
+
+	err = cbor.Unmarshal(raw, &sa)
+	require.NoError(err, "Decode V1 SGX attestation")
+
+	err = sa.ValidateBasic(nil)
+	require.Error(err, "ValidateBasic V1 SGX attestation without PCS support")
+	err = sa.ValidateBasic(&TEEFeatures{})
+	require.Error(err, "ValidateBasic V1 SGX attestation without PCS support")
+	err = sa.ValidateBasic(&TEEFeatures{SGX: TEEFeaturesSGX{PCS: true}})
+	require.NoError(err, "ValidateBasic V1 SGX attestation")
+
+	enc := cbor.Marshal(sa)
+	require.EqualValues(enc, raw, "serialization should round-trip")
+}
+
+func FuzzSGXConstraints(f *testing.F) {
+	// Add some V0 constraints.
+	raw, _ := ioutil.ReadFile("testdata/sgx_constraints_v0.bin")
```

Comment:
does testing.F work with require.noerror? if not can we have it panic here

Reply:
Fixed.